### PR TITLE
Cache for read settings from store

### PR DESCRIPTION
**Pull request by Vendin**
So the driver is correct? If correct, I will finish and add tests.

### CODE REVIEW

**Comment by Vendin** [comment]
@anlutro I also had the option of adding the cache to a higher level and additionally turning it on and off, how do you like this option?

**Comment by Vendin** [comment]
maybe it’s better to insert the cache here regardless of the driver with the option to turn off?

```
	public function load($force = false)
	{
		if (!$this->loaded || $force) {
			$this->data = $this->read();
			$this->loaded = true;
		}
	}
```

**Comment by bweston92** [comment]
Hi @Vendin 
Thanks for your contribution.
Can you please change the spaces to tabs to keep it consistent with the rest of the code base.
I think you should change this to be a CacheSettingStore which will be able to support caching for all stores.

**Comment by Vendin** [comment]
@bweston92 I'll fix it. I want the data to be read only once and from the store, and then taken from the cache.

**Comment by Vendin** [comment]
@bweston92 I waiting for comments and review. Thank you.

**Comment by bweston92** [comment]
Do we want to clear the cache on a write op?

**Comment by Vendin** [comment]
@bweston92 good question. In my case, this is not important. But for other users, perhaps this will be important. I added this possibly optionally in the config. Thank you.

File: tests/functional/DatabaseTest.php [hunk 1]
```diff
@@ -18,6 +18,7 @@ public function setUp()
 			$t->string('key', 64)->unique();
 			$t->string('value', 4096);
 		});
+        parent::setUp();
```
Comment (bweston92) [inline tests/functional/DatabaseTest.php]: tabs not spaces.

File: src/config/config.php [hunk 2]
```diff
@@ -38,7 +38,11 @@
 	'connection' => null,
 	// Name of the table used.
 	'table' => 'settings',
-	// If you want to use custom column names in database store you could 
+	// Cache usage configurations.
+	'enableCache' => false,
+	'forgetCacheByWrite' => true,
+	'cacheTtl' => 15,
+    // If you want to use custom column names in database store you could
```
Comment (bweston92) [inline src/config/config.php]: tabs not spaces.

Comment (Vendin) [reply src/config/config.php]: Done

File: tests/functional/DatabaseTest.php (hunk 1 above)
Comment (Vendin) [reply tests/functional/DatabaseTest.php]: Done

**Comment by bweston92** [comment]
Ok one last thing, can you add some documentation to README or would you prefer I add it?

**Comment by Vendin** [comment]
I will add soon.

**Comment by Vendin** [comment]
@bweston92 added documentation in README. Have a look, please.

**Comment by bweston92** [comment]
Thanks, will tag a release once tested.

**Pull request merged**